### DISCUSSION TITLE
Downloading speed is very slow

### DISCUSSION THREAD

**Discussion by ajhorvath20** [jesec/flood#326] [Q&A]
Hi, 

I am not sure how can I optimize more my `rtorrent.rc` but somehow speed in flood is lower than when comparing with three clients side by side, like in `qbittorrent `and `rtorrentvpn `I am getting `42-43MB/s`, but same file in rflood I am getting `28MB/s` max. Its not going down but its not passing `28MB/s` mark. So I am sure its to do with my config. Please see attached config.

![2](https://user-images.githubusercontent.com/65309559/119991303-398bf900-bfc1-11eb-993a-b483e849a31b.JPG)

My config https://pastebin.com/MJqnSvum 

![image](https://user-images.githubusercontent.com/65309559/119991716-a43d3480-bfc1-11eb-8320-1db77aa465c2.png)

![image](https://user-images.githubusercontent.com/65309559/119991741-af906000-bfc1-11eb-907d-b8c4d1205479.png)

![image](https://user-images.githubusercontent.com/65309559/119991762-b5864100-bfc1-11eb-96ee-27fc95e4a791.png)

Thank you

**Comment by jesec** [comment]
Increase the allowed download slots, total peers and total connections. 

Additionally, make sure that the port is open to incoming connections. rTorrent does not use UPnP to configure port forwarding for you automatically.

There are also other factors that may have unfavorable impacts on rTorrent performance with public torrents: DHT implementation is not optimized, BEP 29 (UTP) and BEP 55 (NAT holepunch) are not implemented.

rTorrent puts greater emphasis on stability, scalability and robust connection. It is great for advanced users, long-term seeders and users of private trackers. Unfortunately, on the other hand, it is not very beginner-friendly, and probably is not a good leech-racer.

**Reply by ajhorvath20** [reply]
Hi jesec, thanks for quick reply. Let me quickly increase the slots and come back to you. 

I'm only using private trackers thats why I left DHT disabled. I only used rtorrentvpn and just recently moved to qbittorrent and seen the client struggling with 600 torrents in the queue so I just wanna keep using rtorrent but I heard really good things about flood so I wanna completely moved to flood. 

I have port forward via vpn provider so thats not the issue. But I cannot confirm with flood if port is active or not or I don't know how to. But I can check port is active in qbittorrent and rtorrentvpn because they have in status bar. 

Thank you again for always prompt help.
